wgpu-core/src/device/resource.rs

```diff
@@ -2926,6 +2916,48 @@ impl Device {
                     stride: vb_state.array_stride,
                 });
             }
+            let mut last_stride = 0;
+            for attribute in vb_state.attributes.iter() {
+                if attribute.offset % attribute.format.size().min(4) != 0 {
+                    return Err(
+                        pipeline::CreateRenderPipelineError::InvalidVertexAttributeOffset {
+                            location: attribute.shader_location,
+                            offset: attribute.offset,
+                        },
+                    );
+                }
+                if attribute.shader_location >= self.limits.max_vertex_attributes {
+                    return Err(
+                        pipeline::CreateRenderPipelineError::TooManyVertexAttributes {
+                            given: attribute.shader_location,
+                            limit: self.limits.max_vertex_attributes,
+                        },
+                    );
+                }
+                last_stride = last_stride.max(attribute.offset + attribute.format.size());
+            }
+            let max_stride = if vb_state.array_stride == 0 {
+                self.limits.max_vertex_buffer_array_stride as u64
+            } else {
+                vb_state.array_stride
+            };
+            if last_stride > max_stride {
+                return Err(
+                    pipeline::CreateRenderPipelineError::VertexAttributeStrideTooLarge {
+                        index: i as u32,
+                        given: last_stride as u32,
+                        limit: max_stride as u32,
+                    },
+                );
+            }
```

Comment:
We could do this check for each attribute (that would match the spec even more), which would give us better errors, with the small price of performance.

Reply:
This would be great - it would be a much better error and shouldn't be too expensive.

Reply:
Done in https://github.com/gfx-rs/wgpu/pull/6804/commits/3b56f03705493e849a7013414266c3fafa7a862d